### DISCUSSION TITLE
Do we want consistent behavior between `gleam run` and `gleam test` when invoking in subdirectories?

### DISCUSSION THREAD

**Discussion by prestontw** [gleam-lang/gleam#3624] [Ideas & suggestions]
Hello! Thank you so much for all of the awesome work put into the Gleam Language and the surrounding tools.

I have noticed one small inconsistency between `gleam run` and `gleam test` when invoking from subdirectories: `gleam run` will try parent directories while `gleam test` doesn't seem to do so. The repo I'm working out of is at: https://github.com/prestontw/gladvent-2023/tree/gleam-test-directories. This is for `gleam 1.4.1`.

`gleam run` from repo root:
```
gladvent_2023 on  gleam-test-directories [!] via ❄  impure (nix-shell-env)
❯ gleam run
   Compiled in 0.01s
    Running gladvent_2023.main
Hello from gladvent_2023!
```

`gleam test` from repo root:
```
gladvent_2023 on  gleam-test-directories [!] via ❄  impure (nix-shell-env)
❯ gleam test
   Compiled in 0.01s
    Running gladvent_2023_test.main
..
Finished in 0.007 seconds
2 tests, 0 failures
```

`gleam run` from `src`:
```
gladvent_2023/src on  gleam-test-directories [!] via ❄  impure (nix-shell-env)
❯ gleam run
   Compiled in 0.01s
    Running gladvent_2023.main
Hello from gladvent_2023!
```

`gleam test` from `src`:
```
gladvent_2023/src on  gleam-test-directories [!] via ❄  impure (nix-shell-env)
❯ gleam test
   Compiled in 0.01s
    Running gladvent_2023_test.main

Finished in 0.000 seconds
0 tests
```

I added a `tests` directory in `src` to see if running the current directory's `tests` folder was the reason we didn't want to travel up to the repo root, but it doesn't look like that is being run. Given these tests aren't being run, would we want to make `gleam test` from a subdirectory work similarly as `gleam run`?

**Comment by lpil** [comment]
Hello! Both `gleam run` and `gleam test` do this. The difference you're seeing that the `gleeunit` program you're running in your test main function uses the current working directory as the base for test discovery.

**Reply by prestontw** [reply]
Thank you for clarifying that. In that case, why doesn't running `gleam test` from inside of `src` return any results?

**Reply by lpil** [reply]
I think `gleeunit` looks in `./test` for the test modules

**Reply by prestontw** [reply]
Ahh 🤦 , my subdirectory test folder name is `tests`. Thank you for walking through this with me! Sorry for taking your time. 😞

**Reply by lpil** [reply]
Glad you figured it out, sounds like a very confusing typo!

**Reply by prestontw** [reply]
In the case where we don't find tests in the current directory, would we want to try parent directories?

**Reply by lpil** [reply]
That would be up to whatever test library you decide to use
